File: frontend/src/services/watchlistService.js
import api from './api';

const watchlistService = {
  getWatchlists: () => {
    return api.get('/watchlists');
  },

  getWatchlistById: (id) => {
    return api.get(`/watchlists/${id}`);
  },

  createWatchlist: (userId, name, description) => {
    return api.post('/watchlists', null, { params: { userId, name, description } });
  },

  addStockToWatchlist: (watchlistId, stockId) => {
    return api.post(`/watchlists/${watchlistId}/stocks/${stockId}`);
  },

  removeStockFromWatchlist: (watchlistId, stockId) => {
    return api.delete(`/watchlists/${watchlistId}/stocks/${stockId}`);
  },

  deleteWatchlist: (id) => {
    return api.delete(`/watchlists/${id}`);
  }
};

export default watchlistService;

File: backend/src/main/java/com/stockanalysis/service/WatchListService.java
package com.stockanalysis.service;

import com.stockanalysis.dto.WatchListDTO;

import java.util.List;

public interface WatchListService {
    WatchListDTO createWatchList(Long userId, String name, String description);
    WatchListDTO getWatchListById(Long id);
    List<WatchListDTO> getWatchListsByUser(Long userId);
    void deleteWatchList(Long id);
    WatchListDTO addStockToWatchList(Long watchListId, Long stockId);
    void removeStockFromWatchList(Long watchListId, Long stockId);
}
